modeltranslation_wagtail/models.py

```diff
@@ -0,0 +1,111 @@
+
+    for (app, mod) in mods:
+        # Attempt to import the app's translation module.
+        module = '%s.translation' % app
+        before_import_registry = copy.copy(translator._registry)
+        try:
+            import_module(module)
+        except:
+            # Reset the model registry to the state before the last import as
+            # this import will have to reoccur on the next request and this
+            # could raise NotRegistered and AlreadyRegistered exceptions
+            translator._registry = before_import_registry
+
+            # Decide whether to bubble up this error. If the app just
+            # doesn't have an translation module, we can ignore the error
+            # attempting to import it, otherwise we want it to bubble up.
+            if module_has_submodule(mod, 'translation'):
+                raise
+
+    for module in TRANSLATION_FILES:
+        import_module(module)
+
+    # Based on:
+    # https://github.com/infoportugal/wagtail-modeltranslation/commit/
+    # e506050dd5650bff60c0d964bda7bf59485ded65
+    # After all models being registered the Page or BaseSetting subclasses and
+    # snippets are patched
+    for model in translator.get_registered_models():
+        if issubclass(model, Page) or model in get_snippet_models() \
+                or issubclass(model, BaseSetting) or model.__name__ == 'Site':
+            WagtailTranslator(model)
+
+    # In debug mode, print a list of registered models and pid to stdout.
+    # Note: Differing model order is fine, we don't rely on a particular
+    # order, as far as base classes are registered before subclasses.
+    if DEBUG:
+        try:
+            if sys.argv[1] in ('runserver', 'runserver_plus'):
+                models = translator.get_registered_models()
+                names = ', '.join(m.__name__ for m in models)
+                print('modeltranslation: Registered %d models for translation'
+                      ' (%s) [pid: %d].' % (len(models), names, os.getpid()))
+        except IndexError:
+            pass
+
+
+def handle_translation_registrations(*args, **kwargs):
+    """
+    Ensures that any configuration of the TranslationOption(s) are handled when
+    importing wagtail_modeltranslation.
+
+    This makes it possible for scripts/management commands that affect models
+    but know nothing of wagtail_modeltranslation.
+    """
+    from modeltranslation.settings import ENABLE_REGISTRATIONS
+
+    if not ENABLE_REGISTRATIONS:
+        # If the user really wants to disable this, they can, possibly at their
+        # own expense. This is generally only required in cases where other
+        # apps generate import errors and requires extra work on the user's
+        # part to make things work.
+        return
+
+    # Trigger autodiscover, causing any TranslationOption initialization
+    # code to execute.
+    autodiscover()
+
+
+if django.VERSION < (1, 7):
+    handle_translation_registrations()
```

Comment:
IMO we should not have this compatibility retained... plenty of stuff to support in the future, so in case we can forget the past, that'd be nice :)

Reply:
fair enough, specially after the release of 2.0 :)